cmd/retry.go

```diff
@@ -0,0 +1,63 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	json "github.com/minio/colorjson"
+	"github.com/minio/mc/pkg/probe"
+)
+
+type retryManager struct {
+	retries       int
+	maxRetries    int
+	retryInterval time.Duration
+	shouldStop    bool
+	ctx           context.Context
+}
+
+func newRetryManager(ctx context.Context, retryInterval time.Duration, maxRetries int) *retryManager {
+	return &retryManager{
+		retryInterval: retryInterval,
+		maxRetries:    maxRetries,
+		ctx:           ctx,
+	}
+}
+
+type retryMessage struct {
+	SourceURL string `json:"sourceURL"`
+	TargetURL string `json:"targetURL"`
+	Retries   int    `json:"retries"`
+}
+
+func (r *retryManager) Stop() {
+	r.shouldStop = true
```

Comment:
Maybe use the context for cancelling (derive a `context.CancelFunc`) - that way it can be instant when used together with `<-time.After(...)`

Also, when it is to be used outside `encapsulateWithRetry`,  Go will consider this a data race.

Reply:
Agreed, I will fix that on monday

Reply:
I removed the boolean and implemented a retryCtx with a cancelFunc.